google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1756,31 +1772,90 @@ def default_acl
         #
         # @return [Policy] the current Cloud IAM Policy for this bucket
         #
-        # @example
+        # @example Retrieving a Policy that is implicitly version 1:
         #   require "google/cloud/storage"
         #
         #   storage = Google::Cloud::Storage.new
-        #
         #   bucket = storage.bucket "my-todo-app"
         #
         #   policy = bucket.policy
+        #   policy.version # 1
+        #   puts policy.roles["roles/storage.objectViewer"]
         #
-        # @example Retrieve the latest policy and update it in a block:
+        # @example Retrieving a version 3 Policy using `requested_policy_version`:
         #   require "google/cloud/storage"
         #
         #   storage = Google::Cloud::Storage.new
+        #   bucket = storage.bucket "my-todo-app"
+        #
+        #   policy = bucket.policy requested_policy_version: 3
+        #   policy.version # 3
+        #   puts policy.bindings.find do |b|
+        #     b[:role] == "roles/storage.objectViewer"
+        #   end
         #
+        # @example Updating a Policy that is implicitly version 1:
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
         #   bucket = storage.bucket "my-todo-app"
         #
         #   bucket.policy do |p|
-        #     p.add "roles/owner", "user:owner@example.com"
+        #     p.version # 1
+        #     p.remove "roles/storage.admin", "user:owner@example.com"
+        #     p.add "roles/storage.admin", "user:newowner@example.com"
+        #     p.roles["roles/storage.objectViewer"] = ["allUsers"]
+        #   end
+        #
+        # @example Updating a Policy from version 1 to version 3 by adding a condition:
```

Comment:
Oh, forgot to call out that in order to use condition with a Cloud Storage bucket Uniform Bucket Level Access must be enabled.

Reply:
Added to all examples showing update to version 3.